### PR TITLE
Check logs

### LINKED ISSUE #15: Check logs
**Check logs**
As an administator, I want to be able to view all actions that have been made on the application to monitor for suspicious activities. 

**Acceptance Criteria**

- [ ] Given that I navigate to the the admin page, I must presented with all the actions that have taken place inside the game.
- [ ] I must also be presented with methods to filter these events
- [ ] Then the events must be updated to only show the event that meet my criteria

**Sub-task list**

- [ ] Sub-issue #151: Create a basic admin page with the option to view app history.
- [ ] Sub-issue #152: Set up persistence storage to store all events.
- [ ] Sub-issue #153: Set up middle-ware to log all request information to the database
- [ ] Sub-issue #154: Set up SQL queries to filter the stored data, then send data to the admin
- [ ] Sub-issue #155: Ensure that the admin portal can only be accessed by an actual admin.

Metadata
Labels: user-story, medium

### PR DESCRIPTION
**Pull request by Siyabonga10**
### **PR: Admin monitoring and game ID display**

**What's changed**
Added request monitoring so admins can see what's happening in the game
Made the game ID visible in the waiting room (helps with sharing)
Added tests for all the new stuff
**Details**
The admin page shows HTTP requests and socket events
Game ID shows up right in the waiting room
Added proper test coverage
Kept everything simple - using in-memory storage for now
**Testing**
Everything passes. Added new tests for:

Request logging
Socket events
Admin routes
Edge cases like missing cookies
Let me know if you want me to change anything or if something needs more explanation.

Closes #15

### CODE REVIEW

File: src/public/js/getAllPlayers.js [hunk 1]
```diff
@@ -1,4 +1,12 @@
 document.addEventListener('DOMContentLoaded', () => {
+  const gameId = document.cookie
```
Comment (CraftierLamb599) [inline src/public/js/getAllPlayers.js:2]: Have you discussed this feature with the other group member responsible for sharing the game? This my cause merge conflicts.

Comment (Siyabonga10) [reply src/public/js/getAllPlayers.js:2]: Sharing game was meant to be a link that users can use. This displays the game ID to the user so that the player doesnt have to guess/check cookies to see the game ID. I assume it shouldnt cause any merge conflicts.

File: src/config/express.js [hunk 2]
```diff
@@ -1,13 +1,15 @@
-'use-strict'
+'use strict'
 
 const express = require('express')
 const path = require('path')
 const bodyParser = require('body-parser')
 const cookieParser = require('cookie-parser')
+const { requestLoggerMiddleware } = require('../middleware/requestLogger')
```
Comment (CraftierLamb599) [inline src/config/express.js:7]: Inconsistent with the use of module aliases throughout the code-base

File: src/public/js/getAllPlayers.js (hunk 1 above)
Comment (Siyabonga10) [reply src/public/js/getAllPlayers.js:2]: It should have been on a different branch, but its a small aesthetic change so I kept it o this branch

File: src/config/express.js (hunk 2 above)
Comment (Siyabonga10) [reply src/config/express.js:7]: I will fix that

**Review by CraftierLamb599 — APPROVED** [review]
Code was well written and adheres to common JavaScript practice. No major issues can be found and the test coverage is excellent.

File: src/public/js/getAllPlayers.js (hunk 1 above)
Comment (CraftierLamb599) [reply src/public/js/getAllPlayers.js:2]: I believe when discussed, the sharing would take place with either the code or the link. That is how it is currently described according to the scrum board.
